webapi/src/main/scala/org/knora/webapi/responders/v2/ResourcesResponderV2.scala

```diff
@@ -648,42 +703,63 @@ final case class ResourcesResponderV2(
              case None => ZIO.unit
            }
 
-      // Check if resources are deleted, if so, replace them with DeletedResource
-      deletedResourcesReplaced =
-        apiResponse.resources.map { resource =>
-          resource.deletionInfo match {
-            // Resource deleted -> return DeletedResource instead
-            case Some(_) => resource.asDeletedResource(versionDate)
-            // Resource not deleted -> return resource
-            case None =>
-              // deleted values should be shown -> resource can be returned
-              if (showDeletedValues) resource
-              // deleted Values should not be shown -> replace them with generic DeletedValue
-              else resource.withDeletedValues(versionDate)
-          }
-        }
-      responseWithDeletedResourcesReplaced = apiResponse.copy(resources = deletedResourcesReplaced)
-    } yield responseWithDeletedResourcesReplaced
+    } yield apiResponse
   }
 
   /**
-   * Get the preview of a resource.
+   * Get the preview of a resource with deleted resources replaced.
    *
    * @param resourceIris         the resource to query for.
    * @param withDeleted          indicates if the deleted resource should be returned or not.
    *
-   * @param requestingUser       the the user making the request.
+   * @param requestingUser       the user making the request.
    * @return a [[ReadResourcesSequenceV2]].
    */
   def getResourcePreviewV2(
     resourceIris: Seq[IRI],
     withDeleted: Boolean = true,
     targetSchema: ApiV2Schema,
     requestingUser: User,
+  ): Task[ReadResourcesSequenceV2] =
+
+    for {
+      apiResponse <- getResourcePreview(resourceIris, withDeleted, targetSchema, requestingUser)
+
+      // BR: Deleted resources are to be replaced with DeletedResource.
+      responseWithDeletedResourcesReplaced =
+        apiResponse.resources match {
+          case resourceList =>
+            if (resourceList.nonEmpty) {
+              val resourceListWithDeletedResourcesReplaced = resourceList.map { resource =>
+                resource.deletionInfo match {
+                  case Some(_) => resource.asDeletedResource()
+                  case None    => resource.withDeletedValues()
+                }
+              }
+              apiResponse.copy(resources = resourceListWithDeletedResourcesReplaced)
+            } else {
+              apiResponse
+            }
+        }
+    } yield responseWithDeletedResourcesReplaced
+
+  /**
+   * Get the preview of a resource as it is in the triplestore.
+   *
+   * @param resourceIris   the resource to query for.
+   * @param withDeleted    indicates if the deleted resource should be returned or not.
+   *
+   * @param requestingUser the user making the request.
+   * @return a [[ReadResourcesSequenceV2]].
+   */
+  private def getResourcePreview(
```

Comment:
As with my [comment above ](https://github.com/dasch-swiss/dsp-api/pull/3736/files#r2284955424) I find the nearly same name for different methods confusing. It would help me to rename the above to `getResourcePreviewWithDeletedResource`.